Implement environment variable reading at yaml files

pedrobenetton opened the pull request:
As discussed in the last developer's meeting, me and another developer of LNLS were having trouble having to share the same variables at our configuration repository. Specifically the variable CLIENT_ID from server.yaml cannot be the same for us, since it is used for our SSO authentication and must be different for each IP address (we test our codes in different machines).

This implementation replaces any string that starts with "_ENV_" with the environment variable of that string. This is done after the yaml file has been read.

marcus-oscarsson commented:
Well done !, could you also add some developers doc in the documentation section about how this works and the syntax ?

marcus-oscarsson commented:
Hello @pedrobenetton and happy new year ! :). How did it go with the documentation ? :)

pedrobenetton commented:
Hello @marcus-oscarsson  happy new year! Last days were a little busy, thanks for the comment.
I added a new file in the docs folder, let me know if it needs any  change.

marcus-oscarsson commented on mxcubeweb/config.py:
```diff
@@ -19,10 +19,35 @@
+        if isinstance(value, str) and value.startswith("_ENV_"):
+            env_value = os.getenv(value)
+            if env_value is None:
+                error_msg = f"{value} starts with _ENV_ but was not set in environment"
```
Ok, thats nice :), I guess that one could discuss weather the `_ENV_ ` should be included in the actual environment variable name or not. However, lets keep it like this for now at least its clear from where the variable comes.

Commit:
Implement environment variable reading at yaml files

ConfigLoader now replaces any string that starts with "_ENV_" with the
environment variable of that value. It walk recusrively through all
the data that has been collected from yaml file, assumes this data
is inside lists or dictionaries, and performs the change at the
strings.

The reason for implementing this is because we need multiple
CLIENT_ID variables. In this case, setting _ENV_CLIENT_ID in
the environment and then setting CLIENT_ID: _ENV_CLIENT_ID in
the yaml configuration file works.

Commit:
Add documentation

marcus-oscarsson commented:
Thanks well done ! :)
